### PR TITLE
TST: test for categorical index monotonicity

### PR DESCRIPTION
**Pull request by jreback**
- xref occasional failures here: https://travis-ci.org/MacPython/pandas-wheels/jobs/259998677
- incorrect version detection in #17142 of bottleneck

### CODE REVIEW

File: pandas/core/indexes/category.py [hunk 1]
```diff
@@ -316,10 +316,23 @@ def _engine(self):
...
         return self._engine_type(lambda: self.codes.astype('i8'), len(self))
 
+    # introspection
     @cache_readonly
     def is_unique(self):
         return not self.duplicated().any()
 
+    @property
+    def is_monotonic(self):
+        return self.is_monotonic_increasing
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: why is it needed to add this one? Isn't this in the base class?

**Review by jorisvandenbossche — COMMENTED** [review]
just to check, is this changing behaviour? or just additional tests / faster implementation? (as I don't fully understand the link with the failing tests)

File: pandas/core/indexes/category.py (hunk 1 above)
Comment (jreback) [inline pandas/core/indexes/category.py]: fixed in base class instead.

**Comment by jreback** [comment]
> just to check, is this changing behaviour? or just additional tests / faster implementation? (as I don't fully understand the link with the failing tests)

nothing changed. I *thought* it was doing something incorrectly, but appears to be ok. in py2.7-32 something is going wrong occasionally, not sure what exactly.

**Pull request merged**